python/update_package_properties.py

```diff
@@ -0,0 +1,346 @@
+            "version should include 3 levels of versions consists of numbers  separated with dots. e.g: 10.0.1")
+
+
+@parameter_validation
+def is_tag(tag: str):
+    if tag is None or not tag:
+        raise ValueError("tag  should be non-empty and should not be None")
+    if not re.match(r"v\d+\.\d+\.\d+$", tag):
+        raise ValueError(
+            "tag  should start with v and should include 3 levels of versions consists of numbers " +
+            "separated with dots. e.g: v10.0.1")
+
+
+@parameter_validation
+def is_email(email: str):
+    if not string_utils.is_email(email):
+        raise ValueError("Parameter is not in email format")
+
+
+@parameter_validation
+def is_project_changelog_header(header: str):
+    if header is None or not header:
+        raise ValueError("header should be non-empty and should not be None")
+    if not re.match(r"^### \w+\sv\d+\.\d+\.\d+\s\(\w+\s\d+,\s\d+\)\s###$", header):
+        raise ValueError(
+            f"changelog header is in invalid format. Actual:{header} Expected: ### citus v8.3.3 (March 23, 2021) ### ")
+
+
+class ChangelogParams:
+    __latest_changelog: str = ""
+    __project_version: str = ""
+    __project_name: str = ""
+    __fancy: bool = False
+    __fancy_version_number: int
+    __microsoft_email: str = ""
+    __name_surname: str = ""
+    __changelog_date: datetime = datetime.now()
+
+    def get_latest_changelog(self) -> str:
+        return self.__latest_changelog
+
+    def set_latest_changelog(self, param: str):
+        self.__latest_changelog = param
+        return self
+
+    def get_project_name(self) -> str:
+        return self.__project_name
+
+    def set_project_name(self, param: non_empty(non_blank(str))):
+        self.__project_name = param
+        return self
+
+    def get_project_version(self) -> str:
+        return self.__project_version
+
+    def set_project_version(self, param: is_version(str)):
+        self.__project_version = param
+        return self
+
+    def get_fancy(self) -> bool:
+        return self.__fancy
+
+    def set_fancy(self, param: bool):
+        self.__fancy = param
+        return self
+
+    def get_fancy_version_number(self) -> int:
+        return self.__fancy_version_number
+
+    def set_fancy_version_number(self, param: non_negative(int)):
+        self.__fancy_version_number = param
+        return self
+
+    def get_microsoft_email(self) -> str:
+        return self.__microsoft_email
+
+    def set_microsoft_email(self, param: is_email(str)):
+        self.__microsoft_email = param
+        return self
+
+    def get_name_surname(self) -> str:
+        return self.__name_surname
+
+    def set_name_surname(self, param: non_empty(non_blank(str))):
+        self.__name_surname = param
+        return self
+
+    def get_changelog_date(self) -> datetime:
+        return self.__changelog_date
+
+    def set_changelog_date(self, param: datetime):
+        self.__changelog_date = param
+        return self
```

Comment:
Using a dataclass makes this a lot less code, it automatically creates an `__init__` method for you:

```suggestion
@dataclass
class ChangelogParams:
    latest_changelog: str = ""
    project_version: str = ""
    project_name: str = ""
    fancy: bool = False
    fancy_version_number: int
    microsoft_email: str = ""
    name_surname: str = ""
    changelog_date: datetime = datetime.now()
```

Docs: https://docs.python.org/3/library/dataclasses.html

Reply:
Fixed

Reply:
https://github.com/citusdata/tools/commit/be5591732f65aef151977d4523eb5ba560f5d5fd